web-console/4.0-designs/storage/Expand Storage Cluster/Expand Storage Cluster.md

```diff
@@ -0,0 +1,71 @@
+# Expand OCS Cluster
+
+## Supported Deployments for OCS 4.2
+
+- AWS (IPI & UPI)
+- VMware (UPI only)
+- RHHI.Next (Bare metal)
+
+## Expand Resource Wireframes 
+We added to the storage section a new entity for “Storage Resources” which lists all the different storage platforms that are applicable or currently configured on OpenShift:
+
+## Add Capacity
+To add capacity, the flow will be decided based on the platform OpenShift is deployed on:
+
+- **Add Capacity** - add capacity by creating underlying storage devices/volumes (in the cloud, i.e. AWS/cloud)
+- **Add Storage Devices** - use existing nodes & add available storage devices/volumes to select from (for UPI/unmanaged, i.e. VMware / Baremetal)
+
+### Add Capacity (AWS/cloud)
+![Storage Resources List](img/Expand-Storage-01.png) 
+In the resource kebab menu, the user will find an action called “Add Capacity” clicking on that will open the add modal.
+
+Note: only Add Capacity is needed for OCS 4.2. Add Node and Add Storage Device would be post OCS 4.2.
+
+![Add Capacity Modal](img/Expand-Storage-02.png) 
+In the “Add Capacity” modal, the user needs to decide on how much capacity he wants to add, and which storage class this request will be made against.
+
+Note: The storage class presented should be filtered to only existing storage classes for the provisioners of the infrastructure OpenShift is deployed on, e.g. AWS or other cloud providers.
+- Storage Class “Info tip”- The Storage Class will be used to request storage from the underlying infrastructure to create the backing persistent volumes that will be used to provide the OpenShift Container Storage (OCS) service.
+
+![Expanding Status](img/Expand-Storage-04.png)
+While the expansion is in progress, we will present an “Expanding” status if possible.
+Note: “Expanding" would cover Add Node, Add Capacity, and Expand/Grow Capacity (in the future).
+
+![Online Status](img/Expand-Storage-05.png)
+When the expansion process is completed, the status will be changed back to online and a toast notification will be presented.
+ 
+
+### Add Storage Devices (For OCS 4.3)
+
+Admin needs to select devices on the node/host, so they can be converted into CR, which in turn becomes the PVs.
+
+![Add Storage Device](img/Expand-Storage-14.png)
+Clicking on Add devices will open the “Add Storage Device” Form. 
+
+![Add Storage Device Form](img/Expand-Storage-15.png)
+![Add Storage Device Form](img/Expand-Storage-16.png)
+After selecting the devices, clicking add will attach them to the OCS service. 
+
+Note: based on the infrastructure the devices list will show different columns: 
+
+- **For local storage for a baremetal node, device list should include** 
+    
+    Device Name, Device UUID, Type, Capacity/Size, Model, Status
+
+
+## Add Node
```

Comment:
@yuvalgalanti  Let's mention this as Add Node (> OCS 4.2) as this is optional for now as the alternative is for user to label the nodes in the Node List (which is what is planned for 4.2).

Reply:
@yuvalgalanti We may want to remove the "Add Node" section entirely as this may evolve / change post 4.2.